src/server/dflycmd.cc

```diff
@@ -318,41 +305,13 @@ void DflyCmd::Flow(CmdArgList args, RedisReplyBuilder* rb, ConnectionContext* cn
     // In this flow the master and the registered replica where synced from the same master.
     if (last_master_id && data && data->id == *last_master_id) {
       ++ServerState::tlocal()->stats.psync_requests_total;
-      std::vector<std::string_view> lsn_str_vec = absl::StrSplit(*last_master_lsn, '-');
-      if (lsn_str_vec.size() != data.value().last_journal_LSNs.size()) {
-        return rb->SendError(facade::kSyntaxErr);  // Unexpected flow. LSN vector of same master
-                                                   // should be the same size on all replicas.
-      }
-      std::vector<LSN> lsn_vec;
-      lsn_vec.reserve(lsn_str_vec.size());
-      for (string_view lsn_str : lsn_str_vec) {
-        int64_t value;
-        if (!absl::SimpleAtoi(lsn_str, &value)) {
-          return rb->SendError(facade::kInvalidIntErr);
-        }
-        lsn_vec.push_back(value);
-      }
-
-      if (IsLSNDiffBellowThreshold(data.value().last_journal_LSNs, lsn_vec)) {
-        sync_type = "PARTIAL";
-        flow.start_partial_sync_at = sf_->journal()->GetLsn();
-        VLOG(1) << "Partial sync requested from LSN=" << flow.start_partial_sync_at.value()
-                << " and is available. (current_lsn=" << sf_->journal()->GetLsn() << ")";
+      auto flow_lsn = ParseLsnVec(*last_master_lsn, data->last_journal_LSNs.size(), flow_id, rb);
+      if (!flow_lsn) {
+        return;  // ParseLsnVec replies in case of error
       }
+      MaybePartialSync(*flow_lsn, &sync_type, &flow);
```

Comment:
sorry, now I looked at what this function do and I think its interface is still akward.
the reason for this is that you wanted to get rid of the duplicated code but I think it can be done differently.
1. introduce in dflycmd.cc `bool IsInJournal(LSN lsn)` that only does the check
2. both branches here will set `seqid` if IsInJournal returns true and clear `seqid` otherwise.

before `rb->StartArray(2);` line based on seqid status you do all the updates/logs that currently reside in `MaybePartialSync`. `MaybePartialSync` is not needed.